src/main/java/com/microsoft/spring/data/gremlin/conversion/GremlinScriptGraphLiteral.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.spring.data.gremlin.conversion;
+
+import com.microsoft.spring.data.gremlin.common.Constants;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import org.springframework.util.Assert;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@NoArgsConstructor
+public class GremlinScriptGraphLiteral implements GremlinScript<String> {
+
+    private String trimScriptHead(@NonNull String script) {
+        return script.replace(Constants.GREMLIN_SCRIPT_HEAD, "") ;
```

Comment:
I do not know the grammar of gremlin, is there any possibility that the replace will replace unexpected `g.` inside the script?

Reply:
Thanks, As gremlin query will always begin with `g.`, here we can use replacefirst.